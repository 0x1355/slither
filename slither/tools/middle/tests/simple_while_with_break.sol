pragma solidity >=0.4.16 <0.7.0;

contract SimpleFor {
    uint public a;

    function SimpleWhileWithBreak(uint input) public {
        for (uint i = 0; i < input; i++) {
            a = a + 1;
            if (i == 8) {
                a = a * 2;
                break;
            }
        }
    }
}