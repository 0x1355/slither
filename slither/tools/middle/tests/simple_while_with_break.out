OVERLAY_FUNCTION_
    ENTRY_POINT
    EXPRESSION
    BREAK
OVERLAY_FUNCTION_
    ENTRY_POINT
    EXPRESSION
    IF
    OVERLAY
    END_IF
    EXPRESSION
    IF_LOOP
    OVERLAY
SimpleWhileWithBreak
    ENTRY_POINT
    NEW VARIABLE
    BEGIN_LOOP
    IF_LOOP
    OVERLAY
    END_LOOP
